File: 1/9.0/9.py
def main():

    input_string = input("Введіть строку для розгляду: ")
    temp_s = ""
    number_of_words = 0
    start = 0
    last = 0
    sum = 0

    number_of_words = count_words(input_string)

    for i in range(0, number_of_words):
        last = last_index(input_string, start)
        temp_s = return_word(input_string, start, last)
        sum += return_number(temp_s)
        start = last

    print(f"Сума елементів рядка дорівнює: {sum}.")


def return_number(s):
    return int(s)


def return_word(s, start, last):
    result = ""

    for i in range(start, last):
        if (s[i] != ' '):
            result += s[i]
        else:
            continue

    return result


def last_index(s, start):
    size = len(s)
    index = start

    for i in range(start, size):
        if (s[i] == ' '):
            index += 1
            continue
        break

    while (index < size):
        if (s[index] != ' '):
            index += 1
        else:
            break

    return index


def count_words(s):
    size = len(s)
    index = 0
    number_of_words = 0

    while (index < size):
        if (s[index] == ' '):
            index += 1
            continue
        else:
            while (index < size):
                if(s[index] != ' '):
                    index += 1
                else:
                    break

            number_of_words += 1

    return number_of_words


main()

File: 1/9.0/9.cpp
#include <iostream>
#include <Windows.h>
#include <iomanip>
#include <string>

int count_words(std::string);
int last_index(std::string, int);
int return_number(std::string);
std::string return_word(std::string, int, int);

int main()
{
    SetConsoleCP(1251);
    SetConsoleOutputCP(1251);

    std::cout << "Введіть строку для розгляду: ";

    std::string input_string = "";
    std::string temp_s = "";
    int number_of_words = 0;
    int start = 0;
    int last = 0;

    int sum = 0;

    std::getline(std::cin, input_string);
    number_of_words = count_words(input_string);

    for (int i = 0; i < number_of_words; ++i)
    {
        last = last_index(input_string, start);
        temp_s = return_word(input_string, start, last);
        sum += return_number(temp_s);
        start = last;
    }

    std::cout << "Сума елементів рядка дорівнює: " << sum << '.' << std::endl;

    return EXIT_SUCCESS;
}

int return_number(std::string s)
{
    return std::stoi(s);
}


std::string return_word(std::string s, int start, int last)
{
    std::string result = "";

    for (int i = start; i < last; ++i)
    {
        if (s[i] != ' ')
            result += s[i];
        else
            continue;      
    }
    return result;
}


int last_index(std::string s, int start)
{
    int size = std::size(s);
    int index = start;
    
    for (int i = start; i < size; ++i)
    {
        if (s[i] == ' ')
        {
            index++;
            continue;
        }
        break;
    }

    while (s[index] != ' ' && index < size)
        index++;

    return index;
}

int count_words(std::string s)
{
    int size = std::size(s);
    int index = 0;
    int number_of_words = 0;

    while (index < size)
    {
        if (s[index] == ' ' || s[index] == '\0')
        {
            index++;
            continue;
        }
        else
        {
            while (s[index] != ' ' && index < size)
            {
                index++;
            }
            number_of_words++;
        }
    }
    return number_of_words;
}